File: QE-Java Selenium/Activity22.java
package demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity22 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.manage().window().maximize();

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/popups");

        System.out.println("Page Title: " + driver.getTitle());

        driver.findElement(By.xpath("//*[@id='launcher']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id='username']")));

        driver.findElement(By.xpath("//*[@id='username']")).sendKeys("admin");

        driver.findElement(By.xpath("//*[@id='password']")).sendKeys("password");

        System.out.println("Entered username and password");

        driver.findElement(By.xpath("//*[@id='signInForm']/button")).click();

        String confirmationMessage = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("/html/body/div/main/div/div/div/div/div/h2"))).getText();

        System.out.println("Confirmation Message: " + confirmationMessage);

        driver.quit();
    }
}

File: QE-Python Selenium/Activity22.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

driver = webdriver.Firefox()
driver.maximize_window()

wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/popups")

print("Page Title:", driver.title)

driver.find_element(By.XPATH, "//*[@id='launcher']").click()

wait.until(EC.visibility_of_element_located((By.XPATH, "//*[@id='username']")))

driver.find_element(By.XPATH, "//*[@id='username']").send_keys("admin")

driver.find_element(By.XPATH, "//*[@id='password']").send_keys("password")

print("Entered username and password")

driver.find_element(By.XPATH, "//*[@id='signInForm']/button").click()

confirmation_message = wait.until(EC.visibility_of_element_located((By.XPATH, "/html/body/div/main/div/div/div/div/div/h2"))).text

print("Confirmation Message:", confirmation_message)

time.sleep(2)
driver.quit()
